packages/wp-now/src/wp-now.ts

```diff
@@ -0,0 +1,168 @@
+import { PHP, SupportedPHPVersion } from '@php-wasm/node'
+import path from 'path'
+import { SQLITE_FILENAME, SQLITE_PATH, WORDPRESS_ZIPS_PATH } from './constants'
+import { downloadSqlite, downloadWordPress } from './download'
+
+interface WPNowOptions {
+  phpVersion?: SupportedPHPVersion
+  documentRoot?: string
+  absoluteUrl?: string
+}
+
+const DEFAULT_OPTIONS: WPNowOptions = {
+  phpVersion: '8.0',
+  documentRoot: '/var/www/html',
+  absoluteUrl: 'http://127.0.0.1:8881',
+}
+
+function seemsLikeAPHPFile(path) {
+	return path.endsWith('.php') || path.includes('.php/');
+}
+
+export default class WPNow {
+  php: PHP
+  options: WPNowOptions = DEFAULT_OPTIONS
+
+  static async create(options: WPNowOptions = {}): Promise<WPNow> {
+    const instance = new WPNow();
+    await instance.setup(options);
+    return instance;
+  }
+
+  patchFile = (path, callback) => {
+    this.php.writeFile(path, callback(this.php.readFileAsText(path)));
+  }
+
+  async setup(options: WPNowOptions = {}) {
+    this.options = {
+      ...this.options,
+      ...options,
+    }
+    const { phpVersion, documentRoot, absoluteUrl } = this.options
+    this.php = await PHP.load(phpVersion, {
+      requestHandler: {
+        documentRoot,
+        absoluteUrl,
+        isStaticFilePath: (path) => {
+          try {
+              const fullPath = this.options.documentRoot + path;
+              return this.php.fileExists(fullPath)
+                  && !this.php.isDir(fullPath)
+                  && !seemsLikeAPHPFile(fullPath);
+          } catch (e) {
+              console.error(e);
+              return false;
+          }
+        },
+
+      }
+    })
+    this.php.mkdirTree(documentRoot)
+    this.php.chdir(documentRoot)
+    this.php.writeFile(`${documentRoot}/index.php`, `<?php echo 'Hello wp-now!';`)
+  }
+
+  mountWordpress(fileName = 'latest') {
+    const { documentRoot } = this.options
+    const root = path.join(WORDPRESS_ZIPS_PATH, fileName, 'wordpress')
+    this.php.mount({
+      root,
+    }, documentRoot)
+    this.php.writeFile(
+      `${documentRoot}/wp-config.php`,
+      this.php.readFileAsText(`${documentRoot}/wp-config-sample.php`)
+  );
+  this.patchFile(
+      `${documentRoot}/wp-config.php`,
+      (contents) =>
+          `<?php
+          define('WP_HOME', "${this.options.absoluteUrl}");
+          define('WP_SITEURL', "${this.options.absoluteUrl}");
+          ?>${contents}`
+  );
```

Comment:
it seems like @adamziel has been abstracting this into `setupSiteUrl()`

if we're concatenating `<?php` multiple times we'll run into trouble, plus by defining `WP_HOME` twice we might run into PHP crashes

Reply:
For reference: https://github.com/WordPress/wordpress-playground/pull/214/files#diff-cb4f5ed0445ed9dbcd9b8d2dc83033ac0588fac24c7c5c94f8b98f6cdfadeac6

Reply:
Nice addition! 🥳 

In order to use `defineSiteUrl` I would like to accept a custom `documentRoot`.

Here is my PR: https://github.com/WordPress/wordpress-playground/pull/216

Another option would be changing the `wp-now` documentRoot to `/wordpress`